internal/fs/inode/file.go

```diff
@@ -681,17 +681,28 @@ func (f *FileInode) CreateEmptyTempFile() (err error) {
 	return
 }
 
-// writeToBuffer writes the given content to the in-memory buffer.
-func (f *FileInode) writeToBuffer(data []byte, offset int64) (err error) {
+// ensureBufferedWriteHandler ensures that buffered write handler is non nil.
+func (f *FileInode) ensureBufferedWriteHandler() (err error) {
 	// Initialize bufferedWriteHandler if not done already.
 	if f.bwh == nil {
 		f.bwh, err = bufferedwrites.NewBWHandler(f.name.GcsObjectName(), f.bucket, f.writeConfig.BlockSizeMb, f.writeConfig.MaxBlocksPerFile, f.globalMaxBlocksSem)
 		if err != nil {
 			return fmt.Errorf("failed to create bufferedWriteHandler: %w", err)
 		}
 	}
+	return
+}
 
-	err = f.bwh.Write(data, offset)
+// writeToBuffer writes the given content to the in-memory buffer.
+func (f *FileInode) writeToBuffer(data []byte, offset int64) error {
+	if err := f.ensureBufferedWriteHandler(); err != nil {
+		return err
+	}
 
-	return
+	select {
+	case <-f.bwh.SignalUploadFailure():
```

Comment:
can we have this logic inside buffered_write_handler itself. inode shouldn't understand any of buffered_write_handler.go logic. Only invoke and act if error is returned.

Reply:
My bad, this was earlier required when I was falling back to temp file flow from file inode in case of failure but no longer a requirement. I forgot to refactor this. Done now.